### PR TITLE
Document Atomics.waitAsync

### LINKED ISSUE #2005: Document Atomics.waitAsync
`Atomics.waitAsync` was [shipped in Chrome 87](https://groups.google.com/a/chromium.org/g/blink-dev/c/Kprh_FIyj74/m/ZG58hpO1AwAJ). It needs to be [added to BCD](https://github.com/mdn/browser-compat-data/issues/8930) and documented on MDN.

cc @jpmedley 

MDN URL: https://developer.mozilla.org/en-US/docs/Web/JavaScript

### PR DESCRIPTION
**Pull request by Elchi3**
I'm not really an expert on shared memory, but this is at least a minimal page on Atomics.waitAsync.

Spec: https://tc39.es/proposal-atomics-wait-async/
Explainer: https://github.com/tc39/proposal-atomics-wait-async/blob/master/PROPOSAL.md

Fixes https://github.com/mdn/content/issues/2005

### CODE REVIEW

File: files/en-us/web/javascript/reference/global_objects/atomics/waitasync/index.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+```
+
+In the reading thread or in another thread, the memory location 0 is called and the promise can be resolved with `"ok"`.
+
+```js
+Atomics.notify(int32, 0);
+// { async: false, value: "ok" }
+```
+
+If it isn't resolving to "ok", the value in the shared memory location wasn't the expected on (the promise will resolve to "not-equal") or the timeout was reached (the promise will resolve to "time-out").
```
Comment (teoli2003) [inline files/en-us/web/javascript/reference/global_objects/atomics/waitasync/index.md:74]: Nit:
```suggestion
If it isn't resolving to `"ok"`, the value in the shared memory location wasn't the expected  (the promise will resolve to `"not-equal"`) or the timeout was reached (the promise will resolve to `"time-out"`). 
```

**Review by teoli2003 — CHANGES_REQUESTED** [review]

**Comment by teoli2003** [comment]
Is there a bcd PR?

**Comment by Elchi3** [comment]
> Is there a bcd PR?

Yes, I updated all of the SharedArrayBuffer data in https://github.com/mdn/browser-compat-data/pull/16047 and added this there.

**Review by teoli2003 — APPROVED** [review]
👍

**Pull request merged**